src/api/procedures/setTransferRestrictions.ts

```diff
@@ -66,32 +76,59 @@ export async function prepareSetTransferRestrictions(
     context: {
       polymeshApi: {
         tx: { statistics },
+        query: { asset },
         consts,
       },
     },
-    storage: {
-      restrictionsToAdd,
-      restrictionsToRemove,
-      exemptionsToAdd,
-      exemptionsToRemove,
-      occupiedSlots,
-    },
+    storage: { currentRestrictions, occupiedSlots, needStat, currentStats },
+    context,
   } = this;
   const {
     restrictions: { length: newRestrictionAmount },
+    restrictions,
+    type,
+    ticker,
   } = args;
 
-  const restrictionsToAddAmount = restrictionsToAdd.length;
-  const restrictionsToRemoveAmount = restrictionsToRemove.length;
-  const exemptionsToAddAmount = exemptionsToAdd.length;
-  const exemptionsToRemoveAmount = exemptionsToRemove.length;
-
-  if (
-    !restrictionsToAddAmount &&
-    !restrictionsToRemoveAmount &&
-    !exemptionsToAddAmount &&
-    !exemptionsToRemoveAmount
-  ) {
+  const rawTicker = stringToTicker(ticker, context);
+
+  const exemptions: (string | Identity)[] = [];
+
+  let someDifference = restrictions.length !== currentRestrictions.length;
+  const conditions = restrictions
+    .map(r => {
+      if (!someDifference) {
+        someDifference = !currentRestrictions.find(transferRestriction => {
+          if (transferRestriction.isMaxInvestorCount && type === TransferRestrictionType.Count) {
+            const currentCount = u64ToBigNumber(transferRestriction.asMaxInvestorCount);
+            return currentCount.eq((r as CountTransferRestrictionInput).count);
+          } else if (transferRestriction.isMaxInvestorOwnership && type === 'Percentage') {
+            const currentOwnership = permillToBigNumber(transferRestriction.asMaxInvestorOwnership);
+            return currentOwnership.eq((r as PercentageTransferRestrictionInput).percentage);
+          }
+          return false;
+        });
+      }
+      let condition: TransferRestriction;
+      if (type === TransferRestrictionType.Count) {
+        condition = { type, value: (r as CountTransferRestrictionInput).count };
+      } else {
+        condition = { type, value: (r as PercentageTransferRestrictionInput).percentage };
+      }
+
+      const rawCondition = transferRestrictionToPolymeshTransferCondition(condition, context);
+
+      if (r.exemptedIdentities) {
+        r.exemptedIdentities.forEach(e => {
+          exemptions.push(e);
+        });
+      }
+
+      return rawCondition;
+    })
+    .sort() as BTreeSetTransferCondition;
```

Comment:
Maybe the insides should be in storage? I don't like this cast so far away from the variable its affecting